lhs.js

```diff
@@ -15,6 +15,24 @@ function dom(selector) {
     return new DomLhs(selector);
 }
 
+/*
+ * Of all the dom nodes selected by type() or dom(), return only
+ * the fnodes that satisfy all the predicates imposed by calls to
+ * when()
+ */
+function checkPredicates(predicates, fnodes) {
+    const ret = [];
+    fnodes.forEach (function (item) {
+        const result = predicates.every (function (func) {
```

Comment:
`result` is always a suspiciously uninformative name. I think `matchesPredicates` or `doesMatchPredicates` or `fnodeMatches` or `fnodeMatchesPredicates` would make the `if` below read more clearly.